Os/test/ut/directory/DirectoryRules.cpp

```diff
@@ -193,7 +193,7 @@ void Os::Test::Directory::Tester::ReadAllFiles::action(Os::Test::Directory::Test
     // Number of files read should be the number of files in the directory minus the original seek position
     ASSERT_EQ(outFileCount, state.m_filenames.size());
     for (FwSizeType i = 0; i < outFileCount; i++) {
-        ASSERT_TRUE(state.is_valid_filename(std::string(outArray[i].toChar())));
+        ASSERT_TRUE(state.is_valid_filename(std::string(outArray[i].toChar())));  // NOLINT(clang-analyzer-security.ArrayBound)
```

Comment:
I am not sure how to fix this (or if this is a false positive). Please let me know how I can fix the error (or feel free to take over).